crates/proto/src/domain/block.rs

```diff
@@ -153,10 +148,8 @@ impl TryFrom<GetBlockInputsResponse> for BlockInputs {
                 let witness_record: AccountWitnessRecord = entry.try_into()?;
                 Ok((
                     witness_record.account_id,
-                    AccountWitness::new(
-                        witness_record.initial_state_commitment,
-                        witness_record.proof,
-                    ),
+                    AccountWitness::new(witness_record.proof)
+                        .expect("account witness record should contain a valid account witness"),
```

Comment:
We should probably not panic in deserialization code.

While this code is part of the internal API, I'm worried this might get re-used publicly sometime.